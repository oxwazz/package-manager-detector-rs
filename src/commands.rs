pub fn npm_run<'a>(agent: &'a str) -> impl Fn(Vec<&'a str>) -> Vec<&'a str> {
    move |args| match args.len() {
        len if len >= 2 => {
            let mut result = vec![agent, "run", args[0], "--"];
            result.extend(&args[1..]);
            result
        }
        1 => vec![agent, "run", args[0]],
        _ => vec![],
    }
}

#[derive(Debug)]
pub enum VectorPart_ {
    String(String),
    Number(u16),
}

#[derive(Debug)]
pub enum AgentCommandValue<T> {
    Vector(Vec<T>),
    // Function(Box<dyn Fn(Vec<String>) -> Vec<String>>),
    None,
}

pub fn construct_command<T: std::fmt::Debug>(value: AgentCommandValue<T>) {
    dbg!(value);
    todo!()
}

pub struct ReturnResolveCommand {
    pub command: String,
    pub args: Vec<String>,
}

pub fn resolve_command(agent: &str, command: &str, args: Vec<&str>) -> ReturnResolveCommand {
    todo!()
}
