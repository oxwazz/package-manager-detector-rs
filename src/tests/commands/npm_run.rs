#[allow(unused_imports)]
use crate::commands::DynamicCommand;

/// npm_run
#[test]
fn test_npm_single_argument() {
    let npm = DynamicCommand { agent: "npm" };
    let output = npm.run(vec!["test"]);
    insta::assert_debug_snapshot!(output);
}

#[test]
fn test_npm_multiple_arguments() {
    let npm = DynamicCommand { agent: "npm" };
    let output = npm.run(vec!["test", "unit", "--watch"]);
    insta::assert_debug_snapshot!(output);
}

#[test]
fn test_yarn_single_argument() {
    let yarn = DynamicCommand { agent: "yarn" };
    let output = yarn.run(vec!["build"]);
    insta::assert_debug_snapshot!(output);
}

#[test]
fn test_yarn_multiple_arguments() {
    let yarn = DynamicCommand { agent: "yarn" };
    let output = yarn.run(vec!["test", "integration"]);
    insta::assert_debug_snapshot!(output);
}

#[test]
fn test_empty_args() {
    let npm = DynamicCommand { agent: "yarn" };
    let output = npm.run(vec![]);
    insta::assert_debug_snapshot!(output);
}

#[test]
fn deno_run_single_argument() {
    let npm = DynamicCommand { agent: "deno" };
    let output = npm.run(vec!["test"]);
    insta::assert_debug_snapshot!(output);
}

#[test]
fn deno_run_multiple_argument() {
    let npm = DynamicCommand { agent: "deno" };
    let output = npm.run(vec!["arg0", "arg1-0", "arg1-1"]);
    insta::assert_debug_snapshot!(output);
}
