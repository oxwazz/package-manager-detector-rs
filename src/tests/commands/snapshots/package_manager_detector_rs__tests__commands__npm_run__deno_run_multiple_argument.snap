---
source: src/tests/commands/npm_run.rs
expression: output
snapshot_kind: text
---
[
    "deno",
    "run",
    "npm:arg0",
    "arg1-0",
    "arg1-1",
]
